update-doi-status/src/test/java/no/unit/nva/doi/handler/UpdateDoiStatusHandlerTest.java

```diff
@@ -0,0 +1,166 @@
+    }
+
+    @Test
+    void handleRequestWhereRequestedDoiInFutureThenThrowsIllegalStateException() {
+        var eventInputStream = IoUtils.inputStreamFromResources(BAD_EVENT_WITH_DATE_IN_FUTURE);
+
+        var actualException = assertThrows(IllegalStateException.class,
+            () -> handler.handleRequest(eventInputStream, outputStream, context));
+        assertThat(actualException.getMessage(), is(equalTo("Modified doi is in the future, bailing!")));
+    }
+
+    @Test
+    void handleRequestWhereDoiRequestHolderIsNullDueToBadInputThenThrowsRuntimeExceptionContainingMismatchedInputException() {
+        var inputStream = new ByteArrayInputStream(new byte[0]);
+
+        var actualException = assertThrows(RuntimeException.class,
+            () -> handler.handleRequest(inputStream, outputStream, context));
+        assertThat(actualException.getCause(), is(instanceOf(MismatchedInputException.class)));
+        assertThat(actualException.getMessage(), containsString("No content to map due to end-of-input"));
+    }
+
+    @Test
+    void handleRequestWherePayloadNotMatchingDoiRequestHolderPojoThenThrowsIllegalArgumentException2() {
+        var eventInputStream = IoUtils.inputStreamFromResources(BAD_EVENT_WITH_BAD_PAYLOAD_NOT_MATCHING_POJO);
+
+        var actualException = assertThrows(IllegalArgumentException.class,
+            () -> handler.handleRequest(eventInputStream, outputStream, context));
+        assertThat(actualException.getMessage(), is(equalTo(String.format(ERROR_BAD_DOI_UPDATE_HOLDER_FORMAT,
+            NULL_OBJECT))));
+    }
+
+    @Test
+    void handleRequestWhilePublicationServiceFailsToFetchPublicationThenThrowsRuntimeExceptionCausedByApiGatewayException()
+        throws ApiGatewayException {
+        when(publicationService.getPublication(EXAMPLE_PUBLICATION_ID)).thenThrow(ApiIoException.class);
+
+        var eventInputStream = IoUtils.inputStreamFromResources(OK_EVENT);
+
+        var actualException = assertThrows(RuntimeException.class,
+            () -> handler.handleRequest(eventInputStream, outputStream, context));
+        assertThat(actualException.getCause(), is(Matchers.isA(ApiGatewayException.class)));
+    }
+
+    @Test
+    void handleRequestWhilePublicationServiceFailsToUpdatePublicationThenThrowsRuntimeExceptionCausedByApiGatewayException()
+        throws ApiGatewayException, JsonProcessingException {
+        Publication publication = new Builder()
+            .withIdentifier(EXAMPLE_PUBLICATION_ID)
+            .build();
+        when(publicationService.getPublication(EXAMPLE_PUBLICATION_ID)).thenReturn(publication);
+        when(publicationService.updatePublication(eq(EXAMPLE_PUBLICATION_ID), any(Publication.class)))
+            .thenThrow(ApiIoException.class);
+
+        var eventInputStream = IoUtils.inputStreamFromResources(OK_EVENT);
+        GatewayResponse<Problem> objectGatewayResponse = GatewayResponse.fromOutputStream(outputStream);
+        // Hm? What is correct here?
```

Comment:
@axthosarouris Input on this test case? 
(so I can update and fix the others)

Reply:
what I usually do is to not use `thenThrow` but `thenAnswer` and throw an exception with a message. Then I check the exception, the cause and the errror message. 
Also, if you can make the method name shorter by using then methodThrowsException...When.... It is shorter and clearer. While means 'during' and 'then' means that there was an action before but there isn't. It would hold if publicationServiceDidOneThingAndThenTheOther

Reply:
Updated, solving this.

Reply:
Thanks, still exercising on getting `methodName(Returns/Throws/Does)SomethingWhen(Input/Condition)IsSomething` into my fingers.